tests/utils.sh

```diff
@@ -21,3 +76,99 @@ function kill_bg( )
     fi
     echo_run ${SUDO}kill -$SIGNAL $TOKILL
 }
+
+function kill_all_bg( ) {
+  if [ ! -z "$INTERACTIVE_MODE" ] ; then
+    read -p "Press enter to kill all jobs"
+  fi
+  echo "-- Killing background jobs"
+  for PID in $BG_PIDS; do
+    [[ ! -z "$PID" ]] && kill_bg $PID
+  done
+}
+
+function report_start( )
+{
+  NAME=$1
+  printf "* %-60s        " "$NAME ..." | tee -a $BASENAME.testlog;
+}
+
+function report_end_failure( )
+{
+  TIMEOUT=$1
+  if [ -z $TIMEOUT ]; then
+    printf "FAIL (%u seconds)\n" $SECONDS | tee -a $BASENAME.testlog;
+  else
+    printf "FAIL (%u seconds timeout)\n" $TIMEOUT | tee -a $BASENAME.testlog;
```

Comment:
This works perfectly... I confirmed with the commit of 84e07a9. This is great!
```
-- Starting test 07-aes-ccm
* clean ...                                                           OK   (6 seconds)
* compile ...                                                         OK   (20 seconds)
* start ./test-aesccm.native ...                                      OK   (0/30 seconds)
* run ./test-aesccm.native ...                                        FAIL (120 seconds timeout)
* run ./test-aesccm.native ...                                        SKIP
-- Killing background jobs
kill -9 1827
../utils.sh: line 41:  1827 Killed                  $TEST &> $RUNLOG
```

Reply:
Oh yes we've been using that one intensively :D

Reply:
Though just spotted a buggy log output: the SKIP line should have been for "check" not for "run". Committed a fix for that

Reply:
:+1

Reply:
👍